Removing Sync bound on `ConcurrentIter::Item` was a breaking change

`orx-parallel` depends on the `Sync` bounds on `ConcurrentIter::Item`.

As a result `orx-parallel` won't compile anymore with `orx-concurrent-iter` version 2.4.0.

It still compiles with `orx-concurrent-iter = "=2.3.0"`.

Trying to compile `orx-parallel` with version 2.4.0 prints a lot of errors similar to this one:

```log
error[E0277]: `<I as ConcurrentIter>::Item` cannot be shared between threads safely
  --> /home/z/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/orx-parallel-2.4.0/src/using/computational_variants/u_par.rs:78:17
   |
78 |     type Item = I::Item;
   |                 ^^^^^^^ `<I as ConcurrentIter>::Item` cannot be shared between threads safely
   |
   = help: the trait `Sync` is not implemented for `<I as ConcurrentIter>::Item`
note: required by a bound in `ParIterUsing::Item`
  --> /home/z/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/orx-parallel-2.4.0/src/using/u_par_iter.rs:21:23
   |
21 |     type Item: Send + Sync;
   |                       ^^^^ required by this bound in `ParIterUsing::Item`
help: consider further restricting the associated type
   |
76 |     I: ConcurrentIter, <I as ConcurrentIter>::Item: Sync
   |                        +++++++++++++++++++++++++++++++++
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "orx-concurrent-iter"
-version = "2.4.0"
+version = "3.0.0"
 edition = "2024"
 authors = ["orxfun <orx.ugur.arikan@gmail.com>"]
 description = "A thread-safe and ergonomic concurrent iterator trait and efficient lock-free implementations."

